File: Task5.1C/app/src/main/java/com/example/task51c/adapter/PlaylistAdapter.java
package com.example.task51c.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.task51c.R;
import com.example.task51c.model.VideoItem;

import java.util.List;

public class PlaylistAdapter extends RecyclerView.Adapter<PlaylistAdapter.ViewHolder> {
    private List<VideoItem> videoList;

    public PlaylistAdapter(List<VideoItem> videoList) {
        this.videoList = videoList;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvTitle, tvUrl;
        public ViewHolder(View itemView) {
            super(itemView);
            tvTitle = itemView.findViewById(R.id.tvTitle);
            tvUrl = itemView.findViewById(R.id.tvUrl);
        }
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_video, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        VideoItem item = videoList.get(position);
        holder.tvTitle.setText(item.title);
        holder.tvUrl.setText("Video ID: " + item.videoUrl);
    }

    @Override
    public int getItemCount() {
        return videoList.size();
    }
}

File: Task5.1C/app/src/main/java/com/example/itube/PlaylistAdapter.kt
package com.example.itube

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.TextView
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView

class PlaylistAdapter(private val onItemClick: (Video) -> Unit) :
    ListAdapter<Video, PlaylistAdapter.PlaylistViewHolder>(VideoDiffCallback()) {

    class PlaylistViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val titleTextView: TextView = view.findViewById(R.id.video_title)
        val deleteButton: ImageButton = view.findViewById(R.id.delete_button)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PlaylistViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_playlist, parent, false)
        return PlaylistViewHolder(view)
    }

    override fun onBindViewHolder(holder: PlaylistViewHolder, position: Int) {
        val video = getItem(position)
        holder.titleTextView.text = video.title
        holder.itemView.setOnClickListener { onItemClick(video) }
        holder.deleteButton.setOnClickListener {

        }
    }
}

class VideoDiffCallback : DiffUtil.ItemCallback<Video>() {
    override fun areItemsTheSame(oldItem: Video, newItem: Video): Boolean {
        return oldItem.id == newItem.id
    }

    override fun areContentsTheSame(oldItem: Video, newItem: Video): Boolean {
        return oldItem == newItem
    }
} 